src/main/java/com/codingnomads/betty/logic/services/ProbabilityToWinService.java

```diff
@@ -0,0 +1,44 @@
+package com.codingnomads.betty.logic.services;
+
+import com.codingnomads.betty.logic.models.TeamProbabilityToWin;
+import com.codingnomads.betty.logic.models.TeamSentimentScore;
+import org.springframework.stereotype.Service;
+
+@Service
+public class ProbabilityToWinService {
+
+    public TeamProbabilityToWin getProbabilityToWinFromSentimentAnalysis(TeamSentimentScore teamSentimentScore) {
+        if (teamSentimentScore.getScore() == null) {
+            throw new NullPointerException("Team Score is null");
```

Comment:
As well there are 2 paradigms of programming, assuming everything can be null, or trying to program to never have nulls. I prefer trying to program as if never were nulls. So the input of this method, whoever calls this method is responsable of not giving a null to this method. that way you dont have to check here. Anyways this approach you are using is perfectly fine as well

Reply:
Thanks. I didn't know that. I will leave as is then